api/inc/box_config.h

```diff
@@ -79,7 +88,7 @@ UVISOR_EXTERN void const * const public_box_cfg_ptr;
 /* this macro selects an overloaded macro (variable number of arguments) */
 #define __UVISOR_BOX_MACRO(_1, _2, _3, _4, NAME, ...) NAME
 
-#define __UVISOR_BOX_CONFIG(box_name, acl_list, acl_list_count, stack_size, context_size) \
+#define __UVISOR_BOX_CONFIG(box_name, acl_list, acl_list_count, stack_size, context_size, debug_box_ptr) \
```

Comment:
Please add `debug_box_ptr` to the uVisor config instead of the box config, since each box doesn't need their own debug box. The system as a whole will have one debug box, and this is best set in one spot: the uVisor config.

You'll be able to get away without modifying so many macros this way. You can use `g_debug_driver` directly from `uvisor-input.S`.